.github/workflows/binskim.yaml

```diff
@@ -0,0 +1,128 @@
+
+          echo "agent: ${KATA_AGENT_PATH}"
+          echo "shim: ${KATA_SHIM_PATH}"
+
+          # Move kata pod sandboxing binaries to artifacts/vanilla
+          mv "${KATA_AGENT_PATH}" "${KATA_SHIM_PATH}" artifacts/vanilla/
+
+          echo "Building kata confpod binaries"
+          pushd tools/osbuilder/node-builder/azure-linux
+          make clean
+          make package-confpods
+          popd
+
+          TARDEV_SNAPSHOTTER_PATH=$(find src/tardev-snapshotter/ -type f -name "tardev-snapshotter" | head -n 1)
+          OVERLAY_PATH=$(find src/overlay/ -type f -name "kata-overlay" | head -n 1)
+          echo "tardev: ${TARDEV_SNAPSHOTTER_PATH}"
+          echo "overlay: ${OVERLAY_PATH}"
+
+          # Prepare go binaries for binskim
+          pushd src/runtime
+          strip --strip-unneeded containerd-shim-kata-v2
+          popd
+
+          # Move kata confpod binaries to artifacts/confpods
+          mv "${KATA_AGENT_PATH}" "${KATA_SHIM_PATH}" "${TARDEV_SNAPSHOTTER_PATH}" "${OVERLAY_PATH}" artifacts/confpods/
+
+      - name: Run BinSkim on kata pod sandboxing binaries
+        run: |
+          for binary in artifacts/vanilla/*; do
+            echo "Running BinSkim on $binary"
+            binskim analyze "$binary" --level Error --kind "Pass;Fail" > "${binary}_binskim_result"
+          done
+
+      - name: Run BinSkim on kata confpod binaries
+        run: |
+          for binary in artifacts/confpods/*; do
+            echo "Running BinSkim on $binary"
+            binskim analyze "$binary" --level Error --kind "Pass;Fail" > "${binary}_binskim_result"
+          done
+
+      - name: Validate BinSkim results
+        run: |
+          # Validate pod sandboxing binaries
+          for result in artifacts/vanilla/*_binskim_result; do
+            if [ ! -f "$result" ]; then
+              echo "❌ Error: $result was not generated."
+              exit 1
+            fi
+            echo "Validating: pod sandboxing ${result}" 
+            cat "$result"
+
+            if grep -qi "fail" "$result"; then
+              echo "❌ Error: Failures detected in pod sandboxing binary: $result"
+              exit 1
+            fi
+            echo "--------------------------- End-------------------------"
+          done
+          echo "✅ All pod sandboxing binaries passed BinSkim."
+
+          # Validate confpod binaries
+          for result in artifacts/confpods/*_binskim_result; do
+            if [ ! -f "$result" ]; then
+              echo "❌ Error: $result was not generated."
+              exit 1
+            fi
+            echo "Validating: conf pod ${result}" 
+            cat "$result"
+
+            if grep -qi "fail" "$result"; then
+              echo "❌ Error: Failures detected in Confidential Pod binary: $result"
+              exit 1
+            fi
+            echo "--------------------------- End-------------------------"
+          done
+          echo "✅ All confpod binaries passed BinSkim."
```

Comment:
These should be independent steps so one binary failing doesn't block other results. We should probably rework this step using `matrix:`.